lib/demo_elixir_phoenix_web/controllers/page_controller.ex

```diff
@@ -102,25 +165,65 @@ defmodule DemoElixirPhoenixWeb.PageController do
 
     conn = KindeClientSDK.login(conn, client)
 
-    res = KindeClientSDK.get_all_data(conn)
+    user = KindeClientSDK.get_user_detail(conn)
 
-    render(conn, "index.html", response: nil)
+    render(conn, "callback.html", user: user)
   end
 
-  def pkce_callack(conn, _) do
-    {conn, client} = KindeClientSDK.get_token(conn)
+  def register(conn, _) do
+    grant_type = :authorization_code_flow_pkce
+
+    {conn, client} =
+      KindeClientSDK.init(
+        conn,
+        Application.get_env(:kinde_sdk, :domain) |> String.replace("\"", ""),
+        Application.get_env(:kinde_sdk, :pkce_callback_url) |> String.replace("\"", ""),
+        Application.get_env(:kinde_sdk, :frontend_client_id) |> String.replace("\"", ""),
+        Application.get_env(:kinde_sdk, :client_secret) |> String.replace("\"", ""),
+        grant_type,
+        Application.get_env(:kinde_sdk, :pkce_logout_url) |> String.replace("\"", "")
+      )
+
+    conn = KindeClientSDK.register(conn, client)
+
     res = KindeClientSDK.get_all_data(conn)
-    render(conn, "index.html", response: res)
+    render(conn, "callback.html", response: res)
+  end
+
+  def pkce_callack(conn, _) do
+    {conn, _client} = KindeClientSDK.get_token(conn)
+    user = KindeClientSDK.get_user_detail(conn)
+    render(conn, "index.html", user: user)
   end
 
   def token_endpoint(conn, _) do
-    {conn, client} = KindeClientSDK.get_token(conn)
-    res = KindeClientSDK.get_all_data(conn)
-    render(conn, "index.html", response: res)
+    res = nil
+
+    if KindeClientSDK.authenticated?(conn) do
+      KindeClientSDK.get_kinde_client(conn)
+      {conn, _client} = KindeClientSDK.get_token(conn)
+      res = KindeClientSDK.get_all_data(conn)
+      render(conn, "callback.html", response: res)
+    else
+      render(conn, "index.html", response: res)
+    end
   end
 
   def start(conn, _) do
-    render(conn, "index.html", response: nil)
+    grant_type = :authorization_code_flow_pkce
+
+    {conn, _client} =
+      KindeClientSDK.init(
```

Comment:
` KindeClientSDK.init` is reused multiple times, so it can be written in a function and call where it is needed